### ISSUE TITLE
Spike: Research Solana Wallet Support in Masa React and/or Masa App

### ISSUE THREAD

**Issue by H34D** [masa-finance/masa-react#584]
We are planning to deploy smart contracts to solana. To make sure we can actually use the contracts from our frontend with the given infrastructure we must research the different wallet types available for solana. 

## Acceptance Criteria

- [ ] Try different solana wallets that work with ~~rainbow kit~~ ([why?](https://github.com/masa-finance/masa-next-sbt/issues/1444#issuecomment-2269032727)) some library
- [ ] Make decision proposal how to proceed and which wallet to integrate
- [ ] the wallet must support ledger hardware wallets

---

# Output

### Phantom
The most used Solana wallet on the ecosystem is [Phantom](https://phantom.app/). It could be compared with Metamask on its dominance on the market share. Aside from Solana, it has integration with:
- Ethereum
- Polygon
- Bitcoin Taproot
- Bitcoin Native Segwit

### Solflare
There were some competitors back in 2021 to Phantom like Sollet eventually [died off](https://coinledger.io/tools/best-solana-wallet#what-happened-to-sollet-) but one of the oldest that stayed strong in the community is [Solflare](https://solflare.com/). Lost the initial battle with Phantom due to UI/UX but still has a following.

Another thing worth noting is that it is Solana only, not like Phantom.

### Ledger
Although this is a hardware wallet, it has its own wallet application to use across the web. Phantom and Solflare provide integration with Ledger but having Ledger on its own integrated into our apps might be worth considering.

## How to proceed
~~First of all we must update rainbowkit and wagmi to the latest version. See #590~~<br/>
> [!WARNING]
> First of all we must figure out how to connect Solana wallets to our frontend

Based on the previous decision making on which wallets to support, I'm assuming that we would like to offer the most popular one as we are doing with Metamask for the SBT app. Thus my suggestion would be to just focus integrating Phantom since it is the most popular and it's already supported on Rainbowkit ([Phantom's how-to](https://docs.phantom.app/library-integrations/rainbowkit), [Rainbowkit's mention of Phantom](https://www.rainbowkit.com/docs/custom-wallet-list#phantom)).

### Implementation steps (high-level)

1. Integrate Solana into the Masa-SDK
2. Add Solana as a valid chain/network on the Masa-React repo
3. Add Solana to the SupportedNetworks list
4. On the SBT app, add Phantom to the list of `allowedWallets` at `src/layout/Providers.tsx`

**Comment by juanmanso** [comment]
🟢 **[UPDATE]** _Adding Phantom to our list of accepted and available wallets_

First time working on SDK + Masa-React + SBT integration repos. In this case, I added the Phantom wallet but I'm not able to connect as shown on the video below. I think I'm not configuring the connecter well, will follow that path

| <video src='https://github.com/user-attachments/assets/ad2c5160-9080-4716-a47b-4e44bc9c3650'> |
| :--: |

**Comment by H34D** [comment]
> 🟢 **[UPDATE]** _Adding Phantom to our list of accepted and available wallets_
> 
> First time working on SDK + Masa-React + SBT integration repos. In this case, I added the Phantom wallet but I'm not able to connect as shown on the video below. I think I'm not configuring the connecter well, will follow that path

there is no configuration for solana networks on the sdk, therefore we can not connect to them right now. I am working on that right now

**Comment by H34D** [comment]
I think you must fix this with synthetic data somehow

**Comment by juanmanso** [comment]
> I think you must fix this with synthetic data somehow (in the time being)

In other words, mocking the SDK, right?

**Comment by juanmanso** [comment]
Since I'm having trouble figuring it out from just looking and browsing the codebase, I'm gonna look for some PR where we add a new wallet to know how to do it.

> _DISCLAIMER: I'm not focusing on Solana itself since Phantom also works for ETH. When solana gets finally added to the SDK ([SDK#334](https://github.com/masa-finance/masa-sdk/issues/334) we shall test this completely)_

**Comment by H34D** [comment]
> In other words, mocking the SDK, right?

probably yes, always keep in mind. a spike must not necessarly yield code or a PR but rather a solid base for decision making so it can be estimated or implemented easier in the future.

**Comment by juanmanso** [comment]
> > In other words, mocking the SDK, right?
> 
> probably yes, always keep in mind. a spike must not necessarly yield code or a PR but rather a solid base for decision making so it can be estimated or implemented easier in the future.

Okay, so I'll be updating the list of wallets to have in mind in this ticket's description and continue understanding how the integration of said wallets would work

**Comment by juanmanso** [comment]
Below you can find the same output of the spike I added to the Description of this ticket (sharing it here just to freeze it in case it changes).

Moving this to review since I find this work to be done. Nonetheless, I'd continue digging deeper into Masa-React to understand how the actual implementation would look like (due to my lack of context of this side of the application)

<details>
  <summary> Click to toggle expand/collapse </summary>

> # Output
> 
> ## List of most used Solana Wallets
> 
> ### Phantom
> The most used Solana wallet on the ecosystem is [Phantom](https://phantom.app/). It could be compared with Metamask on its dominance on the market share. Aside from Solana, it has integration with:
> - Ethereum
> - Polygon
> - Bitcoin Taproot
> - Bitcoin Native Segwit
> 
> ### Solflare
> There were some competitors back in 2021 to Phantom like Sollet eventually [died off](https://coinledger.io/tools/best-solana-wallet#what-happened-to-sollet-) but one of the oldest that stayed strong in the community is [Solflare](https://solflare.com/). Lost the initial battle with Phantom due to UI/UX but still has a following.
> 
> Another thing worth noting is that it is Solana only, not like Phantom.
> 
> ### Ledger
> Although this is a hardware wallet, it has its own wallet application to use across the web. Phantom and Solflare provide integration with Ledger but having Ledger on its own integrated into our apps might be worth considering.
> 
> 
> ## How to proceed
> Based on the previous decision making on which wallets to support, I'm assuming that we would like to offer the most popular one as we are doing with Metamask for the SBT app. Thus my suggestion would be to just focus integrating Phantom since it is the most popular and it's already supported on Rainbowkit ([Phantom's how-to](https://docs.phantom.app/library-integrations/rainbowkit), [Rainbowkit's mention of Phantom](https://www.rainbowkit.com/docs/custom-wallet-list#phantom)).
> 
> ### Implementation steps (high-level)
> 
> 1. Integrate Solana into the Masa-SDK
> 2. Add Solana as a valid chain/network on the Masa-React repo
> 3. Add Solana to the SupportedNetworks list
> 4. On the SBT app, add Phantom to the list of `allowedWallets` at `src/layout/Providers.tsx`

</details>

**Comment by H34D** [comment]
> First time working on SDK + Masa-React + SBT integration repos. In this case, I added the Phantom wallet but I'm not able to connect as shown on the video below. I think I'm not configuring the connecter well, will follow that path

i rewatched the video and i think its because we are not running the lastest wagmi and viem there is a PR for that that got stale: https://github.com/masa-finance/masa-react/pull/514

**Comment by H34D** [comment]
i have tried it myself and it does not seem to detect the phantom wallet extension at all on chrome in the current version of rainbowkit/wagmi:

![image](https://github.com/user-attachments/assets/1e9398b4-e81f-4245-b196-ec40ab1dc821)

**Comment by H34D** [comment]
@juanmanso have you tested connecting a ledger to the phantom wallet?

**Comment by juanmanso** [comment]
I haven't since I got no Ledger myself 😞

**Comment by H34D** [comment]
> I haven't since I got no Ledger myself 😞

ok, i tested that, it works well with `Phantom`

**Comment by juanmanso** [comment]
Closed with these follow-up tickets:
- #591 
- https://github.com/masa-finance/masa-next-sbt/issues/1444

**Comment by juanmanso** [comment]
From other projects that support both ETH and SOL I found they either:
- Have a selector for network and then wallet ([birdeye.so](https://birdeye.so/), [marginfi](https://app.marginfi.com/), [magiceden.io](http://magiceden.io/)) 
- Use something else that integrates the selection into a single modal that does not differentiate networks ([dextools.io](http://dextools.io/))

Either way solutions are not straight forward as it seemed, so we need to reassess priorities :disappointed:

**Comment by juanmanso** [comment]
As pointed out by @H34D this depends on how we want to handle the UX on the app. Currently we connect the wallet first and then select the network.

With the current findings, we might need to first have the user select the network and then give them the connect button so they select wallets.

For this UX/UI change to be determined, we need to sync with @giovaroma to find a solution on then move forward with implementation.

**Comment by juanmanso** [comment]
> As pointed out by [@H34D](https://github.com/H34D) this depends on how we want to handle the UX on the app. Currently we connect the wallet first and then select the network.
> 
> With the current findings, we might need to first have the user select the network and then give them the connect button so they select wallets.
> 
> For this UX/UI change to be determined, we need to sync with [@giovaroma](https://github.com/giovaroma) to find a solution on then move forward with implementation.

In parallel, I'll look at ways to connect _any_ solana wallet into our frontend (with emphasis in supporting Phantom which is the one we found most popular)

**Comment by obasilakis** [comment]
As per [solana's docs](https://docs.phantom.app/library-integrations/dynamic), I'm trying out [dynamic](https://docs.dynamic.xyz/introduction/welcome). Looks promising so far.

**Comment by obasilakis** [comment]
> As per [solana's docs](https://docs.phantom.app/library-integrations/dynamic), I'm trying out [dynamic](https://docs.dynamic.xyz/introduction/welcome). Looks promising so far.

It looks like it's working fine. The only issue is that it's a [paid service](https://www.dynamic.xyz/pricing). $99 for the first 2k MAUs and $0.05/MAU after that.

**Comment by juanmanso** [comment]
Joining to pair and try a free alternative here

**Comment by juanmanso** [comment]
Made a spike based on [this docs](https://github.com/anza-xyz/wallet-adapter/blob/master/APP.md) integrating [`@solana/wallet-adapter-wallets`](https://www.npmjs.com/package/@solana/wallet-adapter-wallets):

Made it work but it diverges from our own UX on the SBT app:

| <video src='https://github.com/user-attachments/assets/af8970b2-6651-4295-a577-b63ad424d38a'> |
| :--: |

**Comment by juanmanso** [comment]
Summary loom video to show current status (1.5x or more recommended 😄):

https://www.loom.com/share/111851fc49bb446c87eaf1c1de224d7a?sid=81cc0b47-7e23-444b-b4d3-fca3b7c5aec9

**Comment by juanmanso** [comment]
Added blocked label hoping it gives more visibility

**Comment by juanmanso** [comment]
Since it seems that we are gonna move forward with the free option, I'm trying to make the connection work on `masa-react`.

As it can be seen on the image below, the UI seems to be working but when clicking the `Select Wallet` button, nothing happens:

| ![Image](https://github.com/user-attachments/assets/901d9dc4-bf09-4fa0-a610-d40b86c3f828) |
| :--: |

It may be due to the initiation of the providers, where we are passing an empty wallet's array whereas the example passes a `new UnsafeBurnerWalletAdapter`. We are not using it because by using it we break the Storybook receiving the following error message:

| ![Image](https://github.com/user-attachments/assets/e41bbbde-05c9-4c27-a3ed-28c71c7a6060) | 
| :--: |

We are still investigating what's going on with this issue and will update soon

**Comment by juanmanso** [comment]
🟢 **[UPDATE]**

Fixed the webpack errors but still not popping up the Phantom wallet.

First I tried with [this stack overflow suggestion](https://stackoverflow.com/a/64802753) to change the `webpack.config.js`. Tried everything and no success there. Then I asked myself if it might be a storybook issue rather than a global webpack thing (which would come up eventually but I'm testing on Storybook first).

So, I found [this other stack overflow answer](https://stackoverflow.com/a/71569287), that suggested to change the `.storybook/main.js` file which I did and errors are no longer showing.

Now, I have to figure out why Wallet connection is not showing on Phantom
